staging/src/k8s.io/cloud-provider/controllers/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -149,6 +154,12 @@ func (c *CloudNodeLifecycleController) MonitorNodes(ctx context.Context) {
 			continue
 		}
 
+		// skip the node if it was recently created, it may not have propagated in cloud provider yet
+		if time.Since(node.CreationTimestamp.Time) < c.instanceExistsGracePeriod {
```

Comment:
What event will trigger this to be re-evaluated?  Or does it just sit in this state until the next `NodeMonitorPeriod`?

Can you at least comment here about it?

Reply:
It will be checked again after `NodeMonitorPeriod`, yep. I expect on EKS we'll set `instanceExistsGracePeriod` to ~10x nodeMonitorPeriod (default for that is 5s).